r2-testapp/src/main/java/org/readium/r2/testapp/audiobook/AudiobookService.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 Readium Foundation. All rights reserved.
+ * Use of this source code is governed by the BSD-style license
+ * available in the top-level LICENSE file of the project.
+ */
+
+package org.readium.r2.testapp.audiobook
+
+import android.app.PendingIntent
+import android.content.Intent
+import org.readium.r2.navigator.media.MediaService
+import org.readium.r2.shared.AudioSupport
+import org.readium.r2.shared.publication.Locator
+import org.readium.r2.shared.publication.Publication
+import org.readium.r2.shared.publication.PublicationId
+import org.readium.r2.testapp.db.BooksDatabase
+
+@OptIn(AudioSupport::class)
+class AudiobookService : MediaService() {
+
+    private val books by lazy { BooksDatabase(this).books }
+
+    override fun onCurrentLocatorChanged(publication: Publication, publicationId: PublicationId, locator: Locator) {
+        books.saveProgression(locator, publicationId.toLong())
+    }
+
+    override val navigatorActivityIntent: PendingIntent?
+        get() {
+            val intent = Intent(this, AudiobookActivity::class.java).apply {
+                addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+            }
+            return PendingIntent.getActivity(this, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT)
```

Comment:
To make it easier for Android 12 compatibility, add `PendingIntent.FLAG_IMMUTABLE` as well.
```suggestion
            return PendingIntent.getActivity(this, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE)
```

Reply:
@stevenzeck Do you have more details about this change? I don't really see the link with Android 12 from the documentation of `FLAG_IMMUTABLE`

Reply:
https://developer.android.com/about/versions/12/behavior-changes-12#pending-intent-mutability